x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/EnrichPolicy.java

```diff
@@ -43,16 +44,18 @@
         args -> {
             return new EnrichPolicy(
                 (String) args[0],
-                (QuerySource) args[1],
-                (String) args[2],
+                (String) args[1],
+                (QuerySource) args[2],
                 (String) args[3],
-                (List<String>) args[4],
-                (String) args[5]
+                (String) args[4],
+                (List<String>) args[5],
+                (String) args[6]
             );
         }
     );
 
     static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME);
```

Comment:
I think this should be an optional parameter. The put transport action should set the name as it can be derived from the put policy request class. We only want to make setting the name in the url path required.

Reply:
Removed, now in `NamedPolicy`